server/controller/src/main/java/ai/starwhale/mlops/schedule/SwTaskSchedulerImpl.java

```diff
@@ -67,27 +67,30 @@ public void schedule(Task task) {
                 log.info("task {} is canceled, skip schedule", task.getId());
                 return;
             }
-            ContainerSpecification containerSpecification = taskContainerSpecificationFinder.findCs(task);
-            RunSpec runSpec = RunSpec.builder()
-                    .image(containerSpecification.getImage())
-                    .envs(containerSpecification.getContainerEnvs())
-                    .command(containerSpecification.getCmd())
-                    .resourcePool(task.getStep().getResourcePool())
-                    .requestedResources(task.getTaskRequest().getRuntimeResources())
-                    .build();
             RunEntity runEntity = RunEntity.builder()
                     .taskId(task.getId())
-                    .runSpec(runSpec)
                     .logDir(task.getResultRootPath().logDir())
                     .build();
             runMapper.insert(runEntity);
             Run run = Run.builder()
                     .id(runEntity.getId())
                     .taskId(task.getId())
-                    .runSpec(runSpec)
                     .logDir(runEntity.getLogDir())
                     .build();
             task.setCurrentRun(run);
+
+            ContainerSpecification containerSpecification = taskContainerSpecificationFinder.findCs(task);
```

Comment:
Would it be better if the function's parameter was set as Run?
There would be no need to ensure that task's `currentRun` has been assigned value beforehand. Otherwise, there's an implicit constraint where setCurrentRun must be called before using this function.

Reply:
The initializing of run depends on `containerSpecification`, so the parameter is task.